movieApp.js

```diff
@@ -0,0 +1,181 @@
+  starRating.className = "star-rating";
+  const numberOfStars = 5;
+  let stars = [];
+  for (let x = 0; x < numberOfStars; x++) {
+    const star = document.createElement("i");
+    star.className = "rating__star far fa-star";
+    starRating.appendChild(star);
+    stars.push(star);
+  }
+  const starClassActive = "rating__star fas fa-star";
+  const starClassInactive = "rating__star far fa-star";
+  stars.forEach((star, clickedStarIndex) => {
+    star.onclick = () => {
+      stars.forEach((star, starIndex) => {
+        if (starIndex <= clickedStarIndex) {
+          star.className = starClassActive;
+        } else {
+          star.className = starClassInactive;
+        }
+      });
+    };
+  });
+
+  movieContainer.appendChild(img);
+  movieContainer.appendChild(readMoreBtn);
+  movieContainer.appendChild(starRating);
+
+  // Create element to show details about the movie
+  const movieDetails = document.createElement("div");
+  movieDetails.className = "movie-details";
+  movieDetails.style.display = "none";
+  movieDetails.style.color = "white";
+  movieDetails.innerHTML = `
+           <p><strong>Title:</strong> ${movie.title}</p>
+           <p><strong>Description:</strong> ${movie.description}</p>
+           <p><strong>Release Year:</strong> ${movie.movie_year}</p>
+           <p><strong>Director:</strong> ${movie.director}</p>
+           <p><strong>Actors:</strong> ${movie.actors.join(", ")}</p>
+       `;
+  movieContainer.appendChild(movieDetails);
+
+  // Adding the comment Section
+
+  function createCommentSection() {
+    const commentSection = document.createElement("div");
+    commentSection.innerHTML = "Comments: ";
+    commentSection.style.fontWeight = "bold";
+    commentSection.className = "comment-section";
+    const commentInput = document.createElement("input");
+    commentInput.type = "text";
+    commentInput.value = "";
+    commentInput.placeholder = "Add a Comment";
+    commentInput.style.color = "white";
+    const commentButton = document.createElement("button");
+    commentButton.className = "comment-button";
+    commentButton.innerHTML = "Submit";
+    const comments = document.createElement("div");
+    comments.className = "comments";
+    const commentUl = document.createElement("ul");
+
+    commentButton.addEventListener("click", function () {
+      const commentUl = document.createElement("ul");
+      if (commentInput.value) {
+        const comment = document.createElement("li");
+        comment.textContent = commentInput.value;
+        commentUl.appendChild(comment);
+        commentInput.value = "";
+      }
+      comments.appendChild(commentUl);
+    });
+
+    // append
+    commentSection.appendChild(commentInput);
+    commentSection.appendChild(commentButton);
+    commentSection.appendChild(comments);
+    movieDetails.appendChild(commentSection);
+  }
+  createCommentSection();
+
+  readMoreBtn.addEventListener("click", function onclick() {
```

Comment:
function doesn't have to be named to used in event listener, we will cover this on sunday as well :)

Reply:
I am trying to remove onclick and write just anonymous function but it is not working on toggling.
I will try and change the same on next submission.